napari/utils/migrations.py

```diff
@@ -218,3 +218,71 @@ def __init_subclass__(cls, **kwargs):
     _OldClass.__new__.__signature__ = prealloc_signature  # type: ignore [attr-defined]
 
     return _OldClass
+
+
+class _DeprecatingDict(dict[str, Any]):
```

Comment:
My main concern here is that part of our public API is class marked as a private one. Im not sure if it is valid approacg

Reply:
Yeah, agreed. It's the one thing that stopped me from marking this as ready for review.

I think users of `as_layer_data_tuple` are pretty advanced, so I'm thinking about just dropping the `_` and making this public.

What do you think?